include/libdnf/rpm/repo.hpp

```diff
@@ -118,262 +109,349 @@ struct Repo {
+
+    Repo & operator=(Repo && repo) = delete;
+
+    /// Registers a class that implements callback methods (fastest mirror detection, download state, key import).
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.setCallbacks(std::unique_ptr<RepoCB> && callbacks)
+    void set_callbacks(std::unique_ptr<RepoCB> && callbacks);
+
+    /// Verify repo object configuration
+    /// Will throw exception if Repo has no mirror or baseurl set or if Repo type is unsupported.
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.verify()
+    void verify() const;
 
-    Repo & operator =(Repo && repo) = delete;
+    /// Returns pointer to the repository configuration
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getConfig()
+    ConfigRepo * get_config() noexcept;
 
-    void setCallbacks(std::unique_ptr<RepoCB> && callbacks);
+    /// Returns repository id
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getId()
+    const std::string & get_id() const noexcept;
 
-    /**
-    * @brief Verify repo object configuration
-    *
-    * Will throw exception if Repo has no mirror or baseurl set or if Repo type is unsupported.
-    */
-    void verify() const;
-    ConfigRepo * getConfig() noexcept;
-    const std::string & getId() const noexcept;
+    /// Enable the repository
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.enable()
     void enable();
+
+    /// Disable the repository
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.disable()
     void disable();
-    bool isEnabled() const;
-    bool isLocal() const;
-    /**
-    * @brief Initialize the repo with metadata
-    *
-    * Fetches new metadata from the origin or just reuses local cache if still valid.
-    *
-    * @return true if fresh metadata were downloaded, false otherwise.
-    */
+
+    /// Return whether the repository is enabled.
+    /// @return true if enabled
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.isEnabled()
+    bool is_enabled() const;
+
+    /// Return whether the repository is local.
+    /// @return true if local
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.isLocal()
+    bool is_local() const;
+
+    /// Initialize the repo with metadata
+    /// Fetches new metadata from the origin or just reuses local cache if still valid.
+    /// @return true if fresh metadata were downloaded, false otherwise.
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.load()
     bool load();
-    bool loadCache(bool throwExcept);
-    void downloadMetadata(const std::string & destdir);
-    bool getUseIncludes() const;
-    void setUseIncludes(bool enabled);
-    bool getLoadMetadataOther() const;
-    void setLoadMetadataOther(bool value);
-    int getCost() const;
-    int getPriority() const;
-    std::string getCompsFn();  // this is temporarily made public for DNF compatibility
+
+    /// Loads metadata from local cache
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.loadCache(bool throwExcept)
+    bool load_cache(bool throw_except);
```

Comment:
Why argument throw_except is required. Please could you describe more details in description?

Reply:
Sometimes failure is fatal and other times it is expected. So in some case we want exception. But I'm going to rework this. Function will always return status. An caller will generate exception in fatal cases.

Reply:
I added internal method `bool try_load_cache()` that indicate result by return value.